src/config/active-response.c

```diff
@@ -428,10 +600,200 @@ int ReadActiveCommands(XML_NODE node, void *d1, __attribute__((unused)) void *d2
 
     /* Add command to the list */
     if (!OSList_AddData(l1, (void *)tmp_command)) {
-        merror(LIST_ADD_ERROR);
+        if (output == NULL) {
+            merror(LIST_ADD_ERROR);
+        } else {
+            wm_strcat(output, "Error adding nodes to list.", '\n');
+        }
         free(tmp_command);
         return (-1);
     }
 
+    if (output) {
+        if (tmp_command->name) {
+            free (tmp_command->name);
```

Comment:
Please, change it, and use `os_free` that already does that check and we avoid repeating code.